### ISSUE TITLE
Validation Loss Statistics: min=nan, med=nan, mean=nan, max=nan

### ISSUE THREAD

**Issue by sun2009ban** [paultsw/nice_pytorch#2]
Thank you for the code!
When I ran the code on mnist like this `python train.py --dataset mnist`, I got the output  `Validation Loss Statistics: min=nan, med=nan, mean=nan, max=nan` after training for several epochs.
![default](https://user-images.githubusercontent.com/21270819/51151846-4b184e80-18a7-11e9-9893-24afa8b6c618.PNG)

Please help me, I have no idea what the matter is.

**Comment by phongnhhn92** [comment]
I am having the same issue. Any fixes for this?

**Comment by ranery** [comment]
![image](https://user-images.githubusercontent.com/18202368/54808365-f7691d80-4cba-11e9-8597-39e6092b92c3.png)
The model collapse very quickly with exploded loss. I want to figure out why.

**Comment by ranery** [comment]
It seems nothing to do with initialization methods.

**Comment by ranery** [comment]
I found it! the adam parameter beta_2 should never be 0.01 :)

**Comment by ranery** [comment]
> I found it! the adam parameter beta_2 should never be 0.01 :)

Still collapse after few epochs :(

**Comment by ranery** [comment]
![image](https://user-images.githubusercontent.com/18202368/54830231-15527480-4cf3-11e9-8090-a16bb2d2aba9.png)
Seems like collapse to one mode after 2/3,000 iterations, is here anybody can give a reason?

**Comment by leviszhang** [comment]
> ![image](https://user-images.githubusercontent.com/18202368/54830231-15527480-4cf3-11e9-8090-a16bb2d2aba9.png)
> Seems like collapse to one mode after 2/3,000 iterations, is here anybody can give a reason?

The norm of h, i.e., f(x) gets bigger and bigger as training evolves, leading to the torch.exp(h) term approaches to Infinity. I looked into the model parameters and found the norm of them also increased a lot in training. One way to mitigate the issue is by using L1 regularization. It would also help if you use F.softplus to make the logistic prior calculation stable.
